src/klein/test/test_plating.py

```diff
@@ -60,6 +77,231 @@ def enwidget(self, a, b):
         """
         return {"a": a, "b": b}
 
+    @element.widgeted
+    def deferredEnwidget(self, a, b):
+        """
+        Provide some L{Deferred} values for the L{element} template.
+        """
+        return {"a": succeed(a), "b": succeed(b)}
+
+
+@attr.s
+class DeferredValue(object):
+    """
+    A value within a JSON serializable object that is deferred.
+
+    @param value: The value.
+
+    @param deferred: The L{Deferred} representing the value.
+    """
+    value = attr.ib()
+    deferred = attr.ib(attr.Factory(Deferred))
+
+    def resolve(self):
+        """
+        Resolve the L{Deferred} that represents the value with the
+        value itself.
+        """
+        self.deferred.callback(self.value)
+
+
+jsonAtoms = (st.none() |
+             st.booleans() |
+             st.integers() |
+             st.floats(allow_nan=False) |
+             st.text(printable))
+
+
+def jsonComposites(children):
+    """
+    Creates a Hypothesis strategy that constructs composite
+    JSON-serializable objects (e.g., lists).
+
+    @param children: A strategy from which each composite object's
+        children will be drawn.
+
+    @return: The composite objects strategy.
+    """
+    return (st.lists(children) |
+            st.dictionaries(st.text(printable), children) |
+            st.tuples(children))
+
+
+jsonObjects = st.recursive(jsonAtoms, jsonComposites, max_leaves=200)
+
+
+def transformJSONObject(jsonObject, transformer):
+    """
+    Recursively apply a transforming function to a JSON serializable
+    object, returning a new, transformed object.
+
+    @param json_object: A JSON serializable object to transform.
+
+    @param transformer: A one-argument callable that will be applied
+        to each member of the object.
+
+    @return: A transformed copy of C{jsonObject}.
+    """
+
+    def visit(obj):
```

Comment:
This looks like it does what you want, but for future reference, [remap](http://boltons.readthedocs.io/en/latest/iterutils.html#boltons.iterutils.remap) can be useful for these scenarios. If you end up wanting to change the non-atom types, I definitely recommend.